File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    answer = ''
    dic = {}
    for i in participant :
        if i in dic :
            dic[i] += 1
        else :
            dic[i] = 1
            
    for i in completion :
        if i in dic :
            dic[i] -= 1
    
    for x, y in dic.items() :
        if y != 0 :
            answer += x
    return answer

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        Map<String, Integer> map = new HashMap<>();

        for (String p : participant) {
            map.put(p, map.getOrDefault(p, 0) + 1);
        }

        for (String c : completion) {
            Integer value = map.get(c);
            if (--value == 0) {
                map.remove(c);
                continue;
            }
            map.put(c, value);
        }

        for (Map.Entry<String, Integer> entry : map.entrySet()) {
            return entry.getKey();
        }
        return null;
    }
}